xdem/coreg/biascorr.py

```diff
@@ -460,7 +494,12 @@ def _apply_rst(  # type: ignore
 
         # Apply function to get correction (including if binning was done before)
         if self._fit_or_bin in ["fit", "bin_and_fit"]:
-            corr = self._meta["fit_func"](tuple(bias_vars.values()), *self._meta["fit_params"])
+            if isinstance(list(bias_vars.values())[0], da.Array):
+                corr = fit_chunked(
```

Comment:
You should be able to do almost exactly the same right below, replacing `self._meta["fit_func"]` for the line `corr = bin_interpolator(*args)`, and the line `corr = get_perbin_nd_binning(*args)`!